src/services/ProcessorService.js

```diff
@@ -1,42 +1,84 @@
 /**
  * Service for Scanning Submissions
  */
-
+const _ = require("lodash");
 const Joi = require("joi");
 const logger = require("../common/logger");
 const helper = require("../common/helper");
+const config = require("config");
+
+async function handleResult(message) {
+  // Move the file to the appropriate bucket
+  if (message.payload.isInfected) {
+    if (message.payload.quarantineDestinationBucket) {
+      // make sure the quarantine bucket is not in the whitelist
+      if (config.get('WHITELISTED_CLEAN_BUCKETS').includes(message.payload.quarantineDestinationBucket)) {
+        logger.warn(`File ${message.payload.fileName} is infected but the quarantine bucket is whitelisted. Chaning to default quarantine bucket`);
+        message.payload.quarantineDestinationBucket = config.get('aws.DEFAULT_QUARANTINE_BUCKET');
+      }
+      await helper.moveFile(
+        config.get('aws.DMZ_BUCKET'),
+        message.payload.fileName,
+        message.payload.quarantineDestinationBucket,
+        message.payload.fileName
+      );
+      // Update the URL
+      message.payload.url = `https://s3.amazonaws.com/${message.payload.quarantineDestinationBucket}/${message.payload.fileName}`;
+    } else {
+      // throw error about missing quarantine bucket
+      throw new Error(`File ${message.payload.fileName} is infected but no quarantine bucket is configured`);
+    }
+  } else {
+    if (message.payload.cleanDestinationBucket) {
+      await helper.moveFile(
+        config.get('aws.DMZ_BUCKET'),
+        message.payload.fileName,
+        message.payload.cleanDestinationBucket,
+        message.payload.fileName
+      );
+      // Update the URL
+      message.payload.url = `https://s3.amazonaws.com/${message.payload.cleanDestinationBucket}/${message.payload.fileName}`;
+    } else {
+      // throw error about missing clean bucket
+      throw new Error(`File ${message.payload.fileName} is clean but no clean bucket is configured`);
+    }
+  }
+  // Notify the caller
+  if (message.payload.callbackTopic) {
+    message.originator = config.get('KAFKA_ORIGINATOR');
+    message.topic = message.payload.callbackTopic;
+    await helper.postToBusAPI(message);
+  }
+  if (message.payload.callbackUrl) {
+    await helper.postToCallbackURL(_.omit(message.payload, ['callbackUrl', 'callbackTopic', 'cleanDestinationBucket', 'quarantineDestinationBucket']));
```

Comment:
this seems not implemented

Reply:
Oh, shoot! I forgot to commit that 🤦‍♂️ ... lemme commit that